src/gg.regression.unity.bots/Runtime/Scripts/RGOverlay/RGBotManager.cs

```diff
@@ -0,0 +1,236 @@
+using RegressionGames;
+using TMPro;
+using UnityEngine;
+
+public class RGBotManager : MonoBehaviour
+{
+    private static RGBotManager _this;
+
+    [Header("Prefabs")]
+    public List<GameObject> botPrefabs;
+
+    [Header("References")]
+    public RGIconPulse launcherPulse;
+
+    public RGIconPulse recordingPulse;
+
+    public GameObject recordingToolbar;
+
+    public GameObject selectionPanel;
+
+    [SerializeField]
+    private TMP_Dropdown gameObjectsDropdown;
+
+    [SerializeField]
+    private TMP_Dropdown behaviorsDropdown;
+
+    [SerializeField]
+    private GameObject activeBotRoot;
+
+    [SerializeField]
+    private GameObject rgBotEntry;
+
+    public static RGBotManager GetInstance()
+    {
+        return _this;
+    }
+
+    private bool _cvRecording = false;
+    private bool _closeOverlayOnBotStart = true;
+    private GameObject _selectedBotPrefab;
+    private string _selectedBehavior;
+    private IRGBotEntry[] _botEntries;
+
+    private void Awake()
+    {
+        if (_this != null && this.gameObject != _this.gameObject)
+        {
+            // we only want one of us around.. kill the other one
+            Destroy(this.gameObject);
+            return;
+        }
+        _this = this;
+        DontDestroyOnLoad(_this.gameObject);
+
+        var newOverlayFeature = RGSettings.GetOrCreateSettings().GetFeatureStateRecordingAndReplay();
+        if (!newOverlayFeature)
+        {
+            if (recordingToolbar != null)
+            {
+                // disable the toolbar
+                recordingToolbar.SetActive(false);
+            }
+        }
+    }
+
+    private void Start()
+    {
+        gameObjectsDropdown.ClearOptions();
+        behaviorsDropdown.ClearOptions();
+
+        AddPrefabsToDropdown();
+        AddBehaviorsToDropdown();
+
+        // Initialize with "Empty" option selected
+        gameObjectsDropdown.value = 0;
+        behaviorsDropdown.value = 0;
+        _selectedBotPrefab = null;
+        _selectedBehavior = null;
+
+        // Subscribe to the dropdown's onValueChanged event
+        gameObjectsDropdown.onValueChanged.AddListener(
+        delegate { PrefabSelected(gameObjectsDropdown); }
+        );
+        behaviorsDropdown.onValueChanged.AddListener(
+        delegate { BehaviorSelected(behaviorsDropdown); }
+        );
+
+        var bots = FindObjectsOfType<MonoBehaviour>().OfType<IRGBot>();
+        foreach (var bot in bots)
+        {
+            Type botType = bot.GetType();
+            string fullyQualifiedName = botType.AssemblyQualifiedName;
+            string botName = FindBotName(fullyQualifiedName);
+            AddActiveBot(((MonoBehaviour)bot).gameObject, botName);
+        }
+    }
+
+    public void OnOverlayClick()
+    {
+        RGDebug.LogVerbose("Showing RG Overlay Menu");
+        selectionPanel.SetActive(true);
+    }
+
+    public void OnOverlayClosed()
+    {
+        selectionPanel.SetActive(false);
+    }
+
+    public void AddBot()
```

Comment:
This code makes the assumption that `_selectedBehavior` has been assigned by `BehaviourSelected`.  But that doesn't get called unless the user changes the dropdown list.  So if they leave it on the first entry, the `_selectedBehavior` is null and the bot fails to start.

Reply:
Fixed